wandb/apis/public.py

```diff
@@ -3973,6 +3975,42 @@ def ref_url(self):
         )
 
 
+class _ArtifactDownloadLogger:
+    def __init__(
+        self,
+        nfiles: int,
+        clock_for_testing: Callable[[], float] = time.time,
+        termlog_for_testing=termlog,
```

Comment:
if the computer decided to sync time while you're downloading your artifact, time.time could in theory go back, so I'd use `time.monotonic`

```suggestion
        clock_for_testing: Callable[[], float] = time.monotonic,
        termlog_for_testing=termlog,
```

also nit: I'm a bit confused why it's called "for_testing", maybe worth renaming?

Reply:
i mean i get what you mean by "testing", but maybe call it probing or something?

Reply:
Hmm, I have trouble imagining any legitimate reason to override these default values in production code -- I was trying to give them names that emphasized that they should only ever be used in tests, so that if anybody tried to override them in production then a reviewer would ask "what are you doing!?".

Do you think "for_probing" conveys that better than "for_testing"? Or do you think this is a misguided goal?

Reply:
you're right, it's fine as is.